git-sync/main.go

```diff
@@ -66,8 +84,24 @@ func main() {
 	if _, err := exec.LookPath("git"); err != nil {
 		log.Fatalf("required git executable not found: %v", err)
 	}
+
+	if *flUsername != "" && *flPassword != "" {
+		log.Println("creating .netrc file (required for https authentication)")
+		if err := createNetrcFile(*flUsername, *flPassword, *flRepo); err != nil {
+			log.Fatalf("error creating .netrc file: %v", err)
+		}
+	}
+
+	if *flOneTime {
+		if err := syncRepo(*flRepo, *flDest, *flBranch, *flRev, *flDepth); err != nil {
+			log.Fatalf("error syncing repo: %v", err)
+		}
+
+		os.Exit(0)
+	}
+
 	for {
-		if err := syncRepo(*flRepo, *flDest, *flBranch, *flRev); err != nil {
+		if err := syncRepo(*flRepo, *flDest, *flBranch, *flRev, *flDepth); err != nil {
```

Comment:
flags are pointers - this is dereferencing them to values.

On Tue, Feb 23, 2016 at 11:32 AM, Mark Turansky <notifications@github.com>
wrote:

> In git-sync/main.go
> <https://github.com/kubernetes/contrib/pull/493#discussion_r53835710>:
>
> >  	for {
> > -		if err := syncRepo(*flRepo, *flDest, *flBranch, *flRev); err != nil {
> > +		if err := syncRepo(*flRepo, *flDest, *flBranch, *flRev, *flDepth); err != nil {
>
> same with all of these. Does passing flRepo, for example, not work? Why
> pointers?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/contrib/pull/493/files#r53835710>.
>